posthog/ai/providers/openai/openai.py

```diff
@@ -0,0 +1,167 @@
+import time
+import uuid
+from typing import Any, Dict, Optional, Union
+
+try:
+    import openai
+except ImportError:
+    raise ModuleNotFoundError("Please install OpenAI to use this feature: 'pip install openai'")
+
+from posthog.ai.utils import call_llm_and_track_usage, get_model_params
+from posthog.client import Client as PostHogClient
+
+
+class OpenAI:
+    """
+    A wrapper around the OpenAI SDK that automatically sends LLM usage events to PostHog.
+    """
+
+    def __init__(
+        self,
+        posthog_client: PostHogClient,
+        **openai_config: Any,
+    ):
+        """
+        Args:
+            api_key: OpenAI API key.
+            posthog_client: If provided, events will be captured via this client instead
+                            of the global posthog.
+            **openai_config: Any additional keyword args to set on openai (e.g. organization="xxx").
+        """
+        self._openai_client = openai.OpenAI(**openai_config)
+        self._posthog_client = posthog_client
+
+    def __getattr__(self, name: str) -> Any:
+        """
+        Expose all attributes of the underlying openai.OpenAI instance except for the 'chat' property,
+        which is replaced with a custom ChatNamespace for usage tracking.
+        """
+        if name == "chat":
+            return self.chat
+        return getattr(self._openai_client, name)
+
+    @property
+    def chat(self) -> "ChatNamespace":
+        return ChatNamespace(self._posthog_client, self._openai_client)
+
+
+class ChatNamespace:
+    def __init__(self, posthog_client: Union[PostHogClient, Any], openai_client: Any):
+        self._ph_client = posthog_client
+        self._openai_client = openai_client
+
+    @property
+    def completions(self):
+        return ChatCompletions(self._ph_client, self._openai_client)
+
+
+class ChatCompletions:
+    def __init__(self, posthog_client: Union[PostHogClient, Any], openai_client: Any):
+        self._ph_client = posthog_client
+        self._openai_client = openai_client
+
+    def create(
+        self,
+        posthog_distinct_id: Optional[str] = None,
+        posthog_trace_id: Optional[str] = None,
+        posthog_properties: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ):
+        distinct_id = posthog_distinct_id or uuid.uuid4()
```

Comment:
How about UUIDv7?

Reply:
we use 4 in the rest of the repo, so trying to keep it consistant